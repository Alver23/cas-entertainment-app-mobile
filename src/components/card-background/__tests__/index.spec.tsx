// Dependencies
import React from 'react';
import { TouchableOpacity } from 'react-native';
import { shallow, ShallowWrapper } from 'enzyme';

// Components
import CardBackground from '../index';

describe('<CardBackground />', () => {
  let component: ShallowWrapper;

  const props = {
    backgroundImage: 'http://localhost/imgae.png',
    onPress: jest.fn(),
    text: 'fake text',
  };

  beforeEach(() => {
    component = shallow(<CardBackground {...props} />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call the onPress void', () => {
    const onPress = component.find(TouchableOpacity).prop('onPress') as any;

    onPress();
    expect(props.onPress).toHaveBeenCalled();
  });
});
