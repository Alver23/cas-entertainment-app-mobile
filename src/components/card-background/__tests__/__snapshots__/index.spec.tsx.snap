// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardBackground /> should save a snapshot of the component 1`] = `
<ForwardRef
  onPress={[MockFunction]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#186AA5",
      "borderRadius": 12,
      "borderWidth": 1,
      "height": 160,
      "justifyContent": "center",
      "margin": 12,
      "width": 160,
    }
  }
>
  <ImageBackground
    source="http://localhost/imgae.png"
    style={
      Object {
        "alignItems": "center",
        "height": 160,
        "justifyContent": "flex-end",
        "width": 160,
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 32,
        }
      }
    >
      <Themed.Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        fake text
      </Themed.Text>
    </View>
  </ImageBackground>
</ForwardRef>
`;
